src/main/scala/com/modelfabric/sparql/stream/client/Builder.scala

```diff
@@ -67,6 +67,7 @@ object Builder {
 
       val responseMerger = builder.add(Merge[SparqlResponse](2).named("merge.sparqlResponse"))
 
+      // JC: are the broadcast and filter necessary? or it can be just a if else logic when building the flow?
```

Comment:
Yes, because parsing the HttpResponse entity unmarshalling is async and results in a future, whereas looking at the header response code is a synchronous operation. Also these steps may be run in parallel.